Classes/Controller/Slot/Extensionmanager/ProcessActions/UpdateConfigurationFileController.php

```diff
@@ -54,32 +53,14 @@ public function updateConfigurationFile($ajaxParams, $ajaxObject)
         if (empty($extensionKey)) {
             return;
         }
-        $packageManager = Bootstrap::getInstance()->getEarlyInstance(PackageManager::class);
-        $packageManager->scanAvailablePackages();
-        $extensionConfigurationPath = $packageManager->getPackage($extensionKey)->getPackagePath() . 'ext_emconf.php';
-        $EM_CONF = null;
-        if (file_exists($extensionConfigurationPath)) {
-            $_EXTKEY = $extensionKey;
-            include $extensionConfigurationPath;
-        }
-
-        if ($EM_CONF === null || empty($EM_CONF[$extensionKey])) {
-            return;
-        }
-
-        $currentMd5HashArray = ExtensionUtility::getMd5HashArrayForExtension($extensionKey);
-        $EM_CONF[$extensionKey]['_md5_values_when_last_written'] = serialize($currentMd5HashArray);
 
-        $emConfUtility = GeneralUtility::makeInstance(EmConfUtility::class);
-        $extensionData = [
-            'extKey' => $extensionKey,
-            'EM_CONF' => $EM_CONF[$extensionKey],
-        ];
-        $emConfContent = $emConfUtility->constructEmConf($extensionData);
-        GeneralUtility::writeFile($extensionConfigurationPath, $emConfContent);
+        $extensionUtility = GeneralUtility::makeInstance(ExtensionUtility::class);
+        $updated = $extensionUtility->updateConfiguration($extensionKey);
 
-        $ajaxObject->setContentFormat('json');
-        $ajaxObject->addContent('title', $this->translate('title'));
-        $ajaxObject->addContent('message', sprintf($this->translate('message'), $extensionKey));
+        if ($updated) {
```

Comment:
What is return if the configuratio was not updated successfully?

Reply:
Not guilty. The original version just did an early return. However, I now added negative feedback on failure. This should be more verbose in the very unlikely event this fails.

Reply:
That is not true. The early return (without any message) is triggered only, if nothing was changed. With your earlier version, only successful changes would trigger feedback because of the additional if.